src/Microsoft.Health.Fhir.Core/Features/Persistence/BadRequestException.cs

```diff
@@ -27,7 +27,7 @@ public BadRequestException(string errorMessage)
         /// <param name="errorMessages">Collection of error strings that are used to initialize OperationOutcomeIssue collection.
         /// Must be non-null and non-empty</param>
         public BadRequestException(ICollection<string> errorMessages)
-            : base((errorMessages.Count < 2) ? errorMessages.First() : "Multiple bad request errors.")
+            : base((errorMessages == null || errorMessages.Count < 2) ? errorMessages.First() : "Multiple bad request errors.")
```

Comment:
If there is going to be a null check here I think it needs to be handled better. Right now if it is null it will try to get the First element and get a null pointer exception. Without the null check in my mind it was implied that null wasn't an expected value, but with the null check I think we need to handle the possibility of it being null.

Reply:
First() is extension, it can and will check for null and throw ArgumentNullException. https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.first

Reply:
I added null check on purpose, even if constructor is by mistake called with null, it will not crash with null ptr but will instead execute First() extension which will handle null ptr properly.